pkg/apis/gcp/cache/v1alpha1/cloudmemorystore_instance_types.go

```diff
@@ -208,27 +210,67 @@ type CloudMemorystoreInstanceList struct {
 
 // NewCloudMemorystoreInstanceSpec creates a new CloudMemorystoreInstanceSpec
 // from the given properties map.
-func NewCloudMemorystoreInstanceSpec(properties map[string]string) *CloudMemorystoreInstanceSpec {
+func NewCloudMemorystoreInstanceSpec(st CloudMemorystoreInstanceClassSpecTemplate) *CloudMemorystoreInstanceSpec {
```

Comment:
Nope! I waffled between the two and ended up going with `SpecTemplate` because it also includes the `provider` and `reclaimPolicy` info and I didn't know if we would want to potentially move the setting of those fields to this function as opposed to the claim reconciler where they currently reside.